lldb/source/Plugins/Language/Swift/SwiftOptional.cpp

```diff
@@ -158,7 +231,65 @@ SwiftOptional_SummaryProvider_Impl(ValueObject &valobj, Stream &stream,
     std::string buffer;
     oneliner.FormatObject(some.get(), buffer, options);
     stream.Printf("%s", buffer.c_str());
-  }
+  } else
+    no_value = true;
+
+  // Display a few children inline.
+
+  // FIXME: This is a temporary hack and not the right design!
+  //        ValueObjectPrinter should just display our value instead.
+  auto is_optional_class = [](CompilerType type) -> bool {
+    auto ts = type.GetTypeSystem().dyn_cast_or_null<TypeSystemSwiftTypeRef>();
+    if (!ts)
+      return false;
+    if (!ts->IsOptionalType(type.GetOpaqueQualType()))
+      return false;
+    CompilerType argument_type =
+        ts->GetGenericArgumentType(type.GetOpaqueQualType(), 0);
+    return argument_type.GetTypeClass() == eTypeClassClass;
+  };
+
+  std::function<void(ValueObject & v, unsigned depth)> format_children =
+      [&](ValueObject &v, unsigned depth) {
```

Comment:
for a lambda of this size, and being recursive, what do you think of making it a regular function? I say this without knowing how many values it captures, and how that would change the signature.